### PR TITLE
Bugfix: Reuse existing manager and monitor

### PR DESCRIPTION
**Pull request by goedderz**
Fixes stack use after scope error found by ASan.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3586/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3586/

File: tests/Aql/BlockFetcherHelper.cpp [hunk 1]
```diff
@@ -213,11 +213,12 @@ std::pair<ExecutionState, AqlItemMatrix const*> AllRowsFetcherHelper::fetchAllRo
 // - SECTION CONSTFETCHER              -
 // -----------------------------------------
 
-ConstFetcherHelper::ConstFetcherHelper(std::shared_ptr<VPackBuffer<uint8_t>> vPackBuffer)
+ConstFetcherHelper::ConstFetcherHelper(AqlItemBlockManager& itemBlockManager,
+                                       std::shared_ptr<VPackBuffer<uint8_t>> vPackBuffer)
     : ConstFetcher(),
       _vPackBuffer(std::move(vPackBuffer)),
-      _resourceMonitor(),
-      _itemBlockManager(&_resourceMonitor),
```
Comment (graetzer) [inline tests/Aql/BlockFetcherHelper.cpp]: is it really necessary to store references to both, instead of just the itemBlockManager ?

Comment (goedderz) [reply tests/Aql/BlockFetcherHelper.cpp]: Actually, we don't need either.

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3586/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3596/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3639/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3648/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3696/

**Pull request merged**